tracer/test/Datadog.Trace.Security.IntegrationTests/AspNetBase.cs

```diff
@@ -409,7 +409,7 @@ protected IImmutableList<MockSpan> WaitForSpans(MockTracerAgent agent, int expec
                 agent.SpanFilters.Add(s => s.Tags.ContainsKey("http.url") && s.Tags["http.url"].IndexOf(url, StringComparison.InvariantCultureIgnoreCase) > -1);
             }
 
-            var spans = agent.WaitForSpans(expectedSpans, minDateTime: minDateTime);
+            var spans = agent.WaitForSpans(expectedSpans, minDateTime: minDateTime, assertExpectedCount: false);
```

Comment:
We should probably not disable this by default 🤔  It was a big effort to enable it as we had a large number of cases where we were failing and we didn't know it 😬

Reply:
I can reenable it, but the reason for disabling it was that when not all the spans are received, we do not know which ones are missing, because the snapshot difference is skipped, and it is an inconvenience.

Reply:
I seem to remember we came up with a way assert, keep a log of it, but not fail the test? Is it possible to do that?

Reply:
I don't get your point here. If there are fewer spans verify will fail and show it in the snapshot difference

Reply:
>I don't get your point here. If there are fewer spans verify will fail and show it in the snapshot difference

Unfortunately that assumes 
1. Every test is using Verify
2. The test does not _filter_ the spans, before sending them to Verify

As for how to do both, you can create an assertion scope, but it's probably not the right option here. Unfortunately, there's not really a great option right now.